nautobot/core/models/tree_queries.py

```diff
@@ -41,16 +43,24 @@ class Meta:
 
     @property
     def display(self):
-        """By default, TreeModels display their full ancestry for clarity."""
+        """
+        By default, TreeModels display their full ancestry for clarity.
+
+        As this is an expensive thing to calculate, we cache it for a few seconds in the case of repeated lookups.
+        """
         if not hasattr(self, "name"):
             raise NotImplementedError("default TreeModel.display implementation requires a `name` attribute!")
-        display_str = ""
+        cache_key = f"{self.__class__.__name__}.{self.id}.display"
+        display_str = cache.get(cache_key, "")
+        if display_str:
+            return display_str
         try:
-            for ancestor in self.ancestors():
-                display_str += ancestor.name + " → "
+            if self.parent is not None:
+                display_str = self.parent.display + " → "
         except self.DoesNotExist:
             # Expected to occur at times during bulk-delete operations
             pass
         finally:
             display_str += self.name
+            cache.set(cache_key, display_str, 5)
```

Comment:
Would it make sense to cache the display name on the model and update the cache for the instance and all descendants any time the `parent` or `name` fields change? If the parent and name fields don't change often would that be more performant?

Reply:
Cascading cache updates might be something we want to look into for dynamic group reversibility performance but for now, I think this is a valuable add.

If we like the pattern for using Redis for caching it makes it easier to do invalidation but I could also see a JSONField "cache" entry on objects for DB-backed caching:

```
>>> location._cache
{
    "display": ["Grad-grandparent Location", "Grandparent Location", "Parent Location", "Location"]
}
```